nixos/modules/services/web-apps/kimai.nix

```diff
@@ -0,0 +1,218 @@
+        description = lib.mdDoc ''
+          `onlySSL` option for nginx virtualHosts.
+          Make sure to set the `useACMEHost` option.
+        '';
+      };
+    };
+  };
+
+  config = mkIf userCfg.enable {
+    users.users.${kimaiName} = {
+      description = "The Kimai service user";
+      isSystemUser = true;
+      group = kimaiName;
+    };
+    users.groups.${kimaiName} = {};
+
+    services.nginx = mkIf userCfg.nginxProxy.enable {
+      enable = true;
+      recommendedProxySettings = true;
+      recommendedTlsSettings = true;
+      virtualHosts = {
+        ${userCfg.nginxProxy.vHost} = {
+          onlySSL = userCfg.nginxProxy.onlySSL;
+          useACMEHost = userCfg.nginxProxy.useACMEHost;
+          root = "${kimaiDataDir}";
+          extraConfig = "index index.php;";
+          locations."~ /\.ht" = {
+            extraConfig = ''
+              deny all;
+            '';
+          };
+          locations."/" = {
+            tryFiles = "$uri /index.php$is_args$args";
+          };
+          locations."~ ^/index\.php(/|$)" = {
+            extraConfig = ''
+              fastcgi_pass unix:${config.services.phpfpm.pools.${kimaiName}.socket};
+              fastcgi_split_path_info ^(.+\.php)(/.+)$;
+              include ${pkgs.nginx}/conf/fastcgi.conf;
+            '';
+          };
+        };
+      };
+    };
+
+    # this is the service that runs our kimai server
+    services.phpfpm.pools.${kimaiName} = {
+      user = kimaiName;
+      group = kimaiName;
+      settings = {
+        "listen.owner" = mkIf userCfg.nginxProxy.enable config.services.nginx.user;
+        "pm" = "dynamic";
+        "pm.max_children" = 32;
+        "pm.max_requests" = 500;
+        "pm.start_servers" = 2;
+        "pm.min_spare_servers" = 2;
+        "pm.max_spare_servers" = 5;
+        "php_admin_value[error_log]" = "stderr";
+        "php_admin_flag[log_errors]" = true;
+        "catch_workers_output" = true;
+      };
+      phpEnv."PATH" = lib.makeBinPath [ pkgs.php ];
+    };
+
+    services.mysql = mkIf userCfg.enableDatabase {
+      enable = true;
+      package = userCfg.mysqlPackage;
+      ensureUsers = [
+        {
+          name = kimaiName;
+          ensurePermissions = {
+            "kimai_db.*" = "ALL PRIVILEGES";
+          };
+        }
+      ];
+      ensureDatabases = [ "kimai_db" ];
+      #  Requires changing the auth for the kimai to work properly,
+      #  because kimai does not use socket_auth,
+      #  which mysql package uses for new users.
+      initialScript = builtins.toFile "mysql_initial.sql" "alter user kimai@localhost identified with mysql_native_password by ''";
+    };
+
+    systemd.services.phpfpm-kimai = {
+      restartTriggers = [ kimaiDrv ];
+      # these preStart scripts run as root
+      preStart = ''
+        mkdir -p /var/lib/kimai/var/log
+        mkdir -p /var/lib/kimai/var/cache
+        mkdir -p /var/lib/kimai/var/data
+        mkdir -p /var/lib/kimai/var/plugins
```

Comment:
These paths should be configurable so people can host multiple instances of it.

Reply:
More importantly they should be using tmpfiles.d